src/external/parsing.rs

```diff
@@ -110,14 +127,25 @@ fn extract_deps_from_value(dependencies: &mut HashSet<String>, deps: &Value) {
 
 /// Extracts dependencies from PEP 735 [dependency-groups] table.
 /// Handles both direct package names and {include-group = "..."} references.
+/// If `include_groups` contains "all", all groups are processed.
+/// Otherwise, only the specified groups are processed.
```

Comment:
i think to avoid ambiguity with a dependency group called `"all"`, we should only allow `"all"` as a string instead of an item in the array. ie. the python type would be `list[str] | Literal["all"]`.

this will also prevent users from specifying `"all"` with other dependency groups, which would be redundant